File: examples/python/textual/notes_showcase/src/notes_showcase/viewmodels/action_vm.py
"""ActionVM — pure record for one capability-derived action.

Used by :class:`~notes_showcase.viewmodels.capability_actions_vm.CapabilityActionsVM`
to project a focused VM's capability surface into a flat list of
``(label, command)`` tuples bound by the view.
"""

from __future__ import annotations

from dataclasses import dataclass

from vmx.commands.protocols import Command


@dataclass(frozen=True, slots=True)
class ActionVM:
    """Pair of a human-readable label and the command it invokes.

    ``command`` is typed as the :class:`~vmx.commands.protocols.Command`
    protocol so the action-bar can reuse decorated commands such as
    :class:`~vmx.commands.ConfirmationDecoratorCommand` (e.g. the wrapped
    ``NoteVM.delete_command``) — shared delete-command behavior.
    """

    label: str
    command: Command

File: examples/swift/notes-showcase/Sources/NotesShowcaseCore/ViewModels/ActionVM.swift
//
// ActionVM — pure presentation value for a single capability-derived action.
//
// Used by `CapabilityActionsVM` (Task 6) to project a focused VM's capability
// surface into a flat list of (label, command) tuples for the view.
// See spec/14-capabilities.md §4 (capability dispatch).
//
import VMx

/// Pure presentation struct for a single capability-derived action.
///
/// Holds a human-readable `label` and the `Command` to invoke when the
/// associated button / menu item is activated. Constructed by the owning
/// `CapabilityActionsVM` from the focused VM's capability methods.
public struct ActionVM {
    public let label: String
    public let command: Command

    public init(label: String, command: Command) {
        self.label = label
        self.command = command
    }
}
